### PR TITLE
Edit link from show page

### PR DESCRIPTION
**Pull request by mattwr18**
Add first test for edit events from show page

Addresses: https://www.pivotaltracker.com/story/show/156008299

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -26,12 +26,24 @@ def create
...
 
   def show
-    @event = Event.find_by_id(params[:id])
     add_breadcrumb @event.title
     @markers = BuildMarkersWithInfoWindow
                    .with(Event.build_by_coordinates([@event]), self)
   end
 
+  def edit
+    @current_user = current_user
```
Comment (tansaku) [inline app/controllers/events_controller.rb]: I can't immediately see why this is needed - am I missing something? :-)

File: features/events/edit.feature [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    Given the following organisations exist:
+      | name            | description          | address        | postcode | website       |
+      | Cats Are Us     | Animal Shelter       | 34 pinner road | HA1 4HZ  | http://a.com/ |
+      | Office Primer   | Care for the elderly | 64 pinner road | HA1 4HZ  | http://b.com/ |
+    Given the following events exist:
+      | title               | description       | organisation  | start_date          | end_date            |
+      | Open Source Weekend | Good for everyone | Cats Are Us   | 2030-10-20 10:30:14 | 2030-10-20 17:00:00 |
+      | Lazy Weekend        | Also good         | Office Primer | 2055-02-02 08:00:00 | 2055-02-02 17:00:00 |
+
+  Scenario: As a logged in user, I can navigate to the edit page of an event
```
Comment (tansaku) [inline features/events/edit.feature]: these look good, although a future work might be to make them more declarative and less imperative:

https://wiki.saucelabs.com/display/DOCS/Best+Practice%3A+Imperative+v.+Declarative+Testing+Scenarios

**Comment by tansaku** [comment]
@mattwr18 wrote:

> I noticed that the most, if not all, of the events have been created before the validations have been created and therefore do not have `start/end_date`s and if someone tries to edit the events, but doesn't include them it just gets stuck on the edit event page and there are no messages telling the user why their event was not created.... 
> When, as a superadmin, an event is created without the proper dates it just refreshes the page... I guess my question is if there is already a ticket to fix this, and should this be a separate ticket that has nothing to do with the above-referenced ticket?
> My feeling is that it's probably a separate ticket.

yeah, sounds like a problem, and a separate ticket ...

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (mattwr18) [reply app/controllers/events_controller.rb]: I think I was using it in the views, but then changed it? I'll have a closer look, but that is my guess.

**Comment by tansaku** [comment]
@mattwr18 so we're good to go here then I think - did you check in with @vikasvr ?

**Comment by mattwr18** [comment]
@tansaku We paired on this actually! He pointed out that when I was calling `@event.update(event_params)` it is actually saving it so there is no reason to need to have a `if @event.save` when you can just add it before the update call. 
We noticed that when removing the instance variable from the `edit` method above it errored out on the 'form' partial, but it seems there was no reason to specifically make it an instance variable there either? We removed from the partial as well and all the tests are passing. 
Could you let me know if the tests are more declarative as you suggested? Or is there still room for improvement?

**Comment by tansaku** [comment]
@mattwr18 looking good - cukes are pretty solidly declarative

only one note is that the edit one could pass without the edit taking place:

```
  Scenario: As a logged in user, I can edit an event
    Given that I am logged in as any user
    Then I visit the edit page for the event titled "Lazy Weekend"
    When I edit the details for "Lazy Weekend"
    Then I should see "Event has been succesfully updated"  
```
```
    And I should see "Lazier Weekend" (or I should see the edit)
```

it's minor but we're not actually checking the event update is visible - arguable whether checking for "Event has been succesfully updated"  or the actual change is most important ... although @NickSchimek  and I just wrangling a bug where we get the success alert but not the actual change ...

wonderful that @vikasvr and you were able to both work on this :-) thanks so much!

**Comment by mattwr18** [comment]
@tansaku I actually had that in there, but then I got worried about DRYing it up and took it out. I thought the only reason you would see that alert is if the update were saved, but I guess it's flakier if you are not actually change the event and then checking for that specific change. Will change it back. :)

**Pull request merged**